lib/remote_files/fog_store.rb

```diff
@@ -98,5 +95,28 @@ def create_directory
         dir.save
       end
     end
+
+    def store_options(file)
+      store_options =
+        {
+          :body => file.content,
+          :content_type => file.content_type,
+          :key => file.identifier,
+          :public => options[:public],
+          :encryption => options[:encryption]
+        }
+      if file.options[:multipart_chunk_size]
+        chunk_size = file.options[:multipart_chunk_size]
+        store_options[:multipart_chunk_size] = chunk_size
+        raise RemoteFiles::Error.new("Minimum chunk size is #{MULITPART_MIN_SIZE}") if chunk_size < MULITPART_MIN_SIZE
+        if !file.content.respond_to?(:read) || !file.content.respond_to?(:size)
+          raise RemoteFiles::Error.new(':content must be a stream if chunking enabled')
```

Comment:
Well, File and StringIO have size().  Not sure what you'd normally pass in here besides those, and I wanted a clean way to catch the AWS error with the max 10000 parts. 
Did you want it be an optional check? So if the stream responds to 'size' then do the verification?